examples/header_bar/main.rs

```diff
@@ -0,0 +1,46 @@
+mod title_bar;
+
+use gtk::gio;
+use gtk::prelude::*;
+use gtk::CheckButton;
+use gtk::Label;
+use gtk::Stack;
+use gtk::{ApplicationWindow, WindowPosition};
+use title_bar::TitleBar;
+
+fn build_ui(application: &gtk::Application) {
+    let window = ApplicationWindow::new(application);
+
+    window.set_title("HeaderBar example");
+    window.set_position(WindowPosition::Center);
```

Comment:
could have used a builder pattern here similar to others

Reply:
Changed to Builder pattern.